### PR TITLE
fix(cost-analysis-data-table): fix and improve incorrect features

### PR DESCRIPTION
**Pull request by sulmoJ**
### To Reviewers
- [ ] Skip (`style`, `chore`, `ci` ONLY)
- [ ] Not that difficult

### Type of Change
- [ ] New feature
- [x] Bug fixes
- [x] Feature improvement
- [ ] Refactor
- [ ] Others (performance improvement, CI/CD, etc.)

### Affects to
- [ ] Packages
  - [ ] core-lib
  - [ ] mirinae
  - [ ] etc 

- [ ] Apps
  - [ ] storybook
  - [x] web

### Checklist
- Did you check the `lint` and `type`?
- Did you document the changes?
  - Changes in `mirinae` should be reflected in `storybook`.
- Did you test the app after the package changes?

### Description
Updated
- Always display the cost/usage dropdown.
- When querying, fetch the `usage_quantity_sum` or `cost_sum` based on the selected option in the dropdown.
- If the `usage` option is chosen in the cost/usage dropdown, display the "Usage Unit" column only when the `usage_type` is selected in the group_by.

### CODE REVIEW

File: apps/web/src/services/cost-explorer/cost-analysis/modules/CostAnalysisDataTable.vue [hunk 1]
```diff
@@ -98,23 +98,17 @@ const state = reactive({
         let dateFormat = 'YYYY-MM';
         if (costAnalysisPageState.granularity === GRANULARITY.YEARLY) dateFormat = 'YYYY';
         const groupBy = state.isIncludedUsageTypeInGroupBy ? [...costAnalysisPageState.groupBy, 'usage_unit'] : costAnalysisPageState.groupBy;
+        const fields = state.usageTypeAdditionalFilterSelected === USAGE_TYPE_ADDITIONAL_FILTER_MAP.usage ? {
+            usage_quantity_sum: { key: 'usage_quantity', operator: 'sum' },
+        } : {
+            cost_sum: { key: 'cost', operator: 'sum' },
+        };
```
Comment (yuda110) [inline apps/web/src/services/cost-explorer/cost-analysis/modules/CostAnalysisDataTable.vue:105]:
```typescript
const fields = {
  value_sum: {
    key: state.usageTypeAdditionalFilterSelected === USAGE_TYPE_ADDITIONAL_FILTER_MAP.usage ? 'usage_quantity' : 'cost',
    operator: 'sum'
  }
};
```

**Review by yuda110 — APPROVED** [review]
LGTM!!! I left a comment

File: apps/web/src/services/cost-explorer/cost-analysis/modules/CostAnalysisDataTable.vue [hunk 2]
```diff
@@ -98,23 +98,17 @@ const state = reactive({
...
-                },
-                ...(state.isIncludedUsageTypeInGroupBy && {
-                    usage_quantity_sum: {
-                        key: 'usage_quantity',
-                        operator: 'sum',
-                    },
-                }),
-            },
+            fields,
             sort: [{ key: '_total_cost_sum', desc: true }],
```
Comment (yuda110) [inline apps/web/src/services/cost-explorer/cost-analysis/modules/CostAnalysisDataTable.vue:113]: todo: this should be changed to

```typescript
sort: [{ key: '_total_value_sum', desc: true }],
```

**Review by WANZARGEN — APPROVED** [review]
LGTM

File: apps/web/src/services/cost-explorer/cost-analysis/modules/CostAnalysisDataTable.vue [hunk 3]
```diff
@@ -278,31 +273,26 @@ const fieldDescriptionFormatter = (field: DataTableFieldType): string => {
...
-                _usageQuantitySum?.push({ date: now.format(dateFormat), value: null });
+            if (!find(target, { date: now.format(dateFormat) })) {
+                target?.push({ date: now.format(dateFormat), value: 0 });
             }
             now = now.add(1, timeUnit);
         }
-        _costSum = sortBy(_costSum, ['date']);
-        _usageQuantitySum = sortBy(_usageQuantitySum, ['date']);
+        target = sortBy(target, ['date']);
+        const targetKey = (isUsageSelected) ? 'usage_quantity_sum' : 'cost_sum';
```
Comment (yuda110) [inline apps/web/src/services/cost-explorer/cost-analysis/modules/CostAnalysisDataTable.vue:292]:
```suggestion
        const targetKey = 'value_sum';
```

**Review by piggggggggy — APPROVED** [review]
LGTM!

**Pull request merged**